aptos-move/move-examples/fungible_asset/stablecoin/sources/fiat_token.move

```diff
@@ -0,0 +1,420 @@
+        assert!(!is_blacklisted(to), error::permission_denied(EBLACKLISTED_ADDRESS));
+
+        // check if amount is greater than `from` address's balance
+        let from_addr = signer::address_of(from); 
+        assert!(balance_of_primary_store(from_addr) < amount, error::permission_denied(EINSUFFICIENT_BALANCE));
+
+        // transfer the tokens from `from` address to `to` address using transfer ref
+        // transfer() with calls withdraw() and deposit(), emitting two events respectively
+        let metadata_object = get_metadata_object();
+        let from_wallet = primary_fungible_store::primary_store(from_addr, metadata_object);
+        let to_wallet = primary_fungible_store::ensure_primary_store_exists(to, metadata_object);
+        fungible_asset::transfer(from, from_wallet, to_wallet, amount);
+
+        // TODO: do we need to emit an event here?
+    }
+    
+    /// Add a new minter with an allowance or update the minter's allowance.
+    /// Only callable by the master minter.
+    /// Returns true if operation is successful.
+    public(friend) fun configure_minter(master_minter: &signer, minter: address, allowance: u64): bool acquires MasterMinterConfig, FiatTokenAccessInfo {
+        // SECURITY TODO: check if contract is paused
+
+        // SECURITY TODO: check if caller is masterMinter
+        let caller_addr = signer::address_of(master_minter);
+        let master_minter_addr = borrow_global<MasterMinterConfig>(@circle).master_minter;
+        assert!(caller_addr == master_minter_addr, error::permission_denied(ENOT_MASTER_MINTER));
+
+        // SECURITY TODO: check if minter is blacklisted
+        assert!(!is_blacklisted(minter), error::permission_denied(EBLACKLISTED_ADDRESS));
+
+        // retrieve the access info and add a new minter and allowance OR update the minter's allowance if already exists
+        let access_info = borrow_global_mut<FiatTokenAccessInfo>(@circle);
+        smart_table::upsert(&mut access_info.minters, minter, allowance);
+
+        // TODO: emit event
+        true
+    }
+    
+    /// Remove a minter from the minter's list.
+    /// Only callable by the master minter.
+    /// Returns true if operation is successful.
+    public(friend) fun remove_minter(master_minter: &signer, minter: address): bool acquires MasterMinterConfig, FiatTokenAccessInfo {
+        // SECURITY TODO: check if contract is paused
+
+        // SECURITY TODO: check if caller is masterMinter 
+        let caller_addr = signer::address_of(master_minter);
+        let master_minter_addr = borrow_global<MasterMinterConfig>(@circle).master_minter;
+        assert!(caller_addr == master_minter_addr, error::permission_denied(ENOT_MASTER_MINTER));
+
+        // remove the minter from the minter's list
+        let access_info = borrow_global_mut<FiatTokenAccessInfo>(@circle);
+        smart_table::remove(&mut access_info.minters, minter);
+
+        // TODO: emit event
+        true
+    }
```

Comment:
These return true or false.  Should it be soft failing on the aborts, and returning false?

Reply:
Okay, then we have two options:
1. abort if it fails
2. soft abort, and return the boolean

Since, you said it returns a boolean, and aborts, let's remove the boolean return type, and just abort or return nothing.